Bottom: 85444282a24d4b63f8deaaa10114d68939eb5add
Top:    75077b4978b2b236767533ebe09db3b3c2efb427
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-02 17:17:24 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9760ab0..870908d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1154,7 +1154,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
